Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+
+extension WKUserContentController {
+    public func addInDefaultContentWorld(scriptMessageHandler: WKScriptMessageHandler, name: String) {
+        if #available(iOS 14.0, *) {
+            add(scriptMessageHandler, contentWorld: .defaultClient, name: name)
+        } else {
+            add(scriptMessageHandler, name: name)
+        }
+    }
+}
+
+extension WKUserScript {
+    public class func createInDefaultContentWorld(source: String, injectionTime: WKUserScriptInjectionTime, forMainFrameOnly: Bool, sandboxed: Bool = true) -> WKUserScript {
+        if #available(iOS 14.0, *), sandboxed {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly, in: .defaultClient)
+        } else {
+            return WKUserScript(source: source, injectionTime: injectionTime, forMainFrameOnly: forMainFrameOnly)
+        }
+    }
+}
+
+public extension WKWebView {
+    private func generateJavascriptFunctionString(functionName: String, args: [Any]) -> String {
+        let argsJS = args
+          .map { "'\(String(describing: $0).escapeHTML())'" }
```

Comment:
what if function name is 'foo' and args is `['); (() => window.cookie = "evil cookie")(']`? would the arg get executed as JS?

Reply:
`escapeArgs` in the function will handle this, see here: https://github.com/brave/brave-ios/pull/3048/files#diff-ac34228e4eb0680b279aaacb48b70fe7566cd87af1f3e54e91444cf879f172b2R30

Reply:
@jumde doesn't that just call escapeHTML (https://github.com/brave/brave-ios/pull/3048/files#diff-ac34228e4eb0680b279aaacb48b70fe7566cd87af1f3e54e91444cf879f172b2R32) on the js args? it may be possible to construct a JS injection that doesn't use any of the characters escaped in https://github.com/brave/brave-ios/pull/3048/files#diff-ac34228e4eb0680b279aaacb48b70fe7566cd87af1f3e54e91444cf879f172b2R67.

for instance `['); (fn () {userPassword = 7})(']`

Reply:
maybe run `JSON.parse` on each arg and throw an error if it doesn't parse correctly? that will at least ensure that each arg corresponds to a valid JS object.

Reply:
@diracdeltas - The idea behind `escapeArgs` in `evaluateSafeJavascript` is to set it to `true` when we expect user-inputs from the webpage to be passed from the content-scripts to the user-scripts for example in the `PaymentHandler` or `WebAuthn`. 

1. In cases where we expect, values like `); (fn () {userPassword = 7})(` to be used as is, we can set the `escapeArgs` to `false`.

2. In cases where we expect values like `); (fn () {userPassword = 7})(` to be escaped, I can update `escapeHTML` to replace `(` | `)` | `{` | `}` with the corresponding HTML entities. Arguments to `evalSafeJavascript` can be of any type, so JSON.parse won't work for all cases.

Let me know if I'm missing something.

Reply:
@jumde right, i was thinking of case 2 where we want the data to be escaped because it may be from an untrusted source. 

> Arguments to evalSafeJavascript can be of any type, so JSON.parse won't work for all cases.

sorry, i mean run the equivalent of the JSON parsing algorithm on the args in Swift before injecting it as JS. 

the benefit of JSON parsing is to validate that the string output in JS will correspond to a Object, Array, string, number, boolean, or null. so for instance `); (fn () {userPassword = 7})(` and other malicious strings should throw an error.